portfolio/src/main/webapp/script.js

```diff
@@ -63,19 +54,147 @@ function showSlide() {
 
   // Show next slide in 5 seconds.
   const TIMEOUT_MILLISECONDS = 5000;
-  timer = window.setTimeout(nextSlide, TIMEOUT_MILLISECONDS);
+  index = (index + 1) % slides.length;
+  timer = window.setTimeout(showSlide, TIMEOUT_MILLISECONDS);      
 }
 
 /**
- * Fetches a quote from the Office from the server and adds it to the DOM.
+ * Fetches all the comments from the server and add them to comment-list.
  */
-function getOfficeQuote() {
-  fetch('/data').then(response => response.json()).then((data) => {
+function getComments() {
+
+  const countEntry = $("#comment-count").val();
+  const pageEntry = $("#page-number").val();
+
+  // Make get request to get commentCount number of comments.
+  $.get("/data", { count: countEntry, page: pageEntry }, function (data, textStatus, jqXHR) {
+    
+    console.log(data);
+    // Empty the list that will receive the comments.
+    const commentList = document.getElementById("comment-list");
+    commentList.innerHTML="";
+
+    // For each comment, create and append a list element.
+    data.comments.forEach((comment) => {
+      commentList.appendChild(createCommentElement(comment));
+    })
     
-    // data is json of QuotePerson, so reference its attributes
-    const quoteContainer = document.getElementById('quote-container');
-    quoteContainer.innerText = "\"" + data.quote + "\" - " + data.person;
+    // Create new page number options based on new get comments.
+    $("#page-number").empty();
+    $("#page-number").append("<option selected='selected' value='1'>1</option>");
+    
+    let key = 2;
+    while(data.count > (key - 1) * countEntry) {
+      $("#page-number").append($("<option></option>").val(key).text(key));
+      key += 1;
+    }
```

Comment:
can you add a comment for what's going on here?

Reply:
Will do